File: platforms/ios/foundation/Sources/NativeblocksFoundation/modifier/TestTag.swift
import NativeblocksCompiler
import NativeblocksRuntime
import SwiftUI

@Modifier(
    name: "Test Tag",
    keyType: "nativeblocks/test_tag",
    description: "Tags the block for UI tests (Maestro).",
    version: 1,
    versionName: "1"
)
struct TestTag: ViewModifier {

    @ModifierData(description: "The tag used by UI test selectors.", defaultValue: "")
    var tag: String = ""

    @ViewBuilder
    func body(content: Content) -> some View {
        if tag.isEmpty {
            content
        } else {
            content.accessibilityIdentifier(tag)
        }
    }
}

File: platforms/android/foundation/foundation/src/main/java/io/nativeblocks/foundation/modifier/TestTag.kt
package io.nativeblocks.foundation.modifier

import io.nativeblocks.compiler.type.Modifier
import io.nativeblocks.compiler.type.ModifierData
import androidx.compose.ui.Modifier as UiModifier
import androidx.compose.ui.platform.testTag as uiTestTag

@Modifier(
    keyType = "nativeblocks/test_tag",
    name = "Test Tag",
    description = "Tags the block for UI tests (Maestro).",
    version = 1,
    versionName = "1",
)
internal fun testTag(
    @ModifierData(description = "The tag used by UI test selectors.")
    tag: String,
): UiModifier {
    if (tag.isBlank()) return UiModifier
    return UiModifier.uiTestTag(tag)
}
